ci/date-check/src/main.rs

```diff
@@ -35,50 +36,56 @@ impl fmt::Display for Date {
     }
 }
 
-fn make_date_regex() -> Regex {
-    Regex::new(
-        r"(?x) # insignificant whitespace mode
-        <!--\s*
-        [dD]ate:\s*
-        (?P<y>\d{4}) # year
-        -
-        (?P<m>\d{2}) # month
-        \s*-->",
-    )
-    .unwrap()
+fn make_date_regex() -> Vec<Regex> {
+    let patterns = [
+        r"<!--\s+date-check:\s+(\w+)\s+(\d+{4})\s+-->",
+        r"<!--\s+date-check\s+-->\s+(\w+)\s+(\d+{4})",
```

Comment:
I don't think the `+` after `\d` is correct (it wasn't there before either):

```suggestion
        r"<!--\s+date-check:\s+(\w+)\s+(\d{4})\s+-->",
        r"<!--\s+date-check\s+-->\s+(\w+)\s+(\d{4})",
```

Reply:
Actually, could you combine these regexes into one to avoid all this complexity with making multiple passes? It might be a little more permissive wrt accepted input but I think it'd be better overall.

Reply:
I did this with an earlier iteration, but it moved complexity elsewhere, where I had 4 regex groups (instead of just 2). Is that preferable, or is there a better alternative?

Reply:
ok, it's not so bad 2a5b586c24d4bed7b78227d5a5481bbb6b47e5ff